Sources/NIOPosix/SelectableEventLoop.swift

```diff
@@ -58,10 +58,12 @@ struct PooledBuffer: PoolElement {
         let count = Socket.writevLimitIOVectors
         let iovecs = self.buffer.bindMemory(to: IOVector.self, capacity: count)
         let storageRefs = (self.buffer + (count * MemoryLayout<IOVector>.stride)).bindMemory(to: Unmanaged<AnyObject>.self, capacity: count)
-        assert((iovecs >= self.buffer) && (iovecs <= (self.buffer + self.bufferSize)))
-        assert((storageRefs >= self.buffer) && (storageRefs <= (self.buffer + self.bufferSize)))
-        assert((iovecs + count) == storageRefs)
-        assert((storageRefs + count) <= (self.buffer + bufferSize))
+        assert(iovecs >= self.buffer.bindMemory(to: IOVector.self, capacity: count))
+        assert(iovecs <= (self.buffer + self.bufferSize).bindMemory(to: IOVector.self, capacity: count))
+        assert(storageRefs >= self.buffer.bindMemory(to: Unmanaged<AnyObject>.self, capacity: count))
+        assert(storageRefs <= (self.buffer + self.bufferSize).bindMemory(to: Unmanaged<AnyObject>.self, capacity: count))
+        assert(UnsafeMutableRawPointer(iovecs + count) == UnsafeMutableRawPointer(storageRefs))
```

Comment:
Don't bind the memory in these assertions, the duplicate bind makes it hard to reason about. Erase the pointers to raw instead.

Reply:
It does not compile with swift 5.5/5.6, CI works fails...

Reply:
or do you mean having it like
```
        assert(UnsafeMutableRawPointer(iovecs) >= self.buffer)
        assert(UnsafeMutableRawPointer(iovecs) <= (self.buffer + self.bufferSize))
        assert(UnsafeMutableRawPointer(storageRefs) >= self.buffer)
        assert(UnsafeMutableRawPointer(storageRefs) <= (self.buffer + self.bufferSize))
        assert(UnsafeMutableRawPointer(iovecs + count) == UnsafeMutableRawPointer(storageRefs))
        assert(UnsafeMutableRawPointer(storageRefs + count) <= (self.buffer + bufferSize))

```